### PR TITLE
Bulk Update Currencies for Listings

### PR DESCRIPTION
**Pull request by hoffmabc**
This is a very similar API call to the one used to update moderators for all listings.

POST http://localhost:4002/ob/bulkupdatecurrency
Payload: { "currencies":["ZEC", "BTC", "BCH", "LTC"] }

This will update the listings.json as well as ALL listings themselves with the updated currency array.

### CODE REVIEW

**Comment by cpacia** [comment]
Looks good at first glance. I'll take a deeper look and test it later today.

**Comment by jjeffryes** [comment]
It looks like this branch set all the listings to track inventory, zero inventory on 2 of my nodes. I'll try and confirm that.

**Comment by jjeffryes** [comment]
I've verified that calling bulkupdatecurrency will set all of your listings' skus to have a quantity of zero instead of whatever they had before, which sets them to track inventory with an inventory of 0, making them unbuyable.

If they have a coupon, it also removes the `discountCode` entirely, which makes the coupon invalid.

This is a fairly serious bug, in my opinion.

**Comment by hoffmabc** [comment]
The reason this bug occurs is that we perform this bulk update incorrectly.

The way it currently works is: 
- It pulls in a listing off disk
- Update the currencies in the JSON
- Update the listing by sending that to the API

This causes a problem because what's on disk is *not* what actually should get sent to the API. It's missing inventory and coupon information.

I will go ahead and update this code to perform this properly.

**Comment by hoffmabc** [comment]
Ok @jjeffryes it looks like this is now considering the inventory and coupons. It wasn't as bad as I thought it would be. @placer14 we need a double check to make sure we're not doing anything else wrong here as we completely overlooked that it wasn't putting the coupons and inventory in there.

**Comment by jjeffryes** [comment]
This seems to be working as intended now.

**Comment by hoffmabc** [comment]
Great. I just need to apply the other fixes @placer14 suggested.

**Comment by hoffmabc** [comment]
> I think we should also suspend the root hash updater while this bulk update is completing. Otherwise, each individual listing update will trigger the root hash to be updated in the push nodes and create a lot of network chatter. The root hash could be resumed after the bulk update and the final merkle root could be enough to cover all the changes.

This latest commit should defer the publish until everything is completed.

**Comment by hoffmabc** [comment]
@jjeffryes ready for testing again with updates.

@placer14 what else is needed here?

File: core/moderation.go [hunk 1]
```diff
@@ -178,6 +178,79 @@ func (n *OpenBazaarNode) GetModeratorFee(transactionTotal uint64, paymentCoin, c
...
+						break
+					}
+				}
+			}
+
+			n.UpdateListing(sl.Listing, false)
+
+		}
+		return nil
+	}
```
[the remark this answers is not in the document]
Comment (hoffmabc) [reply core/moderation.go]: Ok I believe the 100% coverage on this stuff should cover your requests.

**Comment by hoffmabc** [comment]
Ready for review @placer14

**Comment by hoffmabc** [comment]
GTG @cpacia

**Pull request merged**